src/BinaryParsers/ELFBinary/Dwarf/Read.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.BinaryParsers.Dwarf
+{
+    internal static class Read
+    {
+        // Read attribute value from .debug_info
+        public static List<byte> AttributeValue(List<byte> infoData, ref int index, Attribute attribute, int cuId)
+        {
+            var output = new List<byte>();
+            try
+            {
+                switch (attribute.Form)
+                {
+                    case DW_FORM.Addr:
+                        output.AddRange(infoData.GetRange(index, 4));
+                        index += 4;
+                        break;
+
+                    case DW_FORM.Block2:
+                    {
+                        int numBytes = BitConverter.ToInt16(infoData.GetRange(index, 2).ToArray(), 0);
+                        index += 2;
+                        output.AddRange(infoData.GetRange(index, numBytes));
+                        index += numBytes;
+                        break;
+                    }
+                    case DW_FORM.Block4:
+                    {
+                        int numBytes = BitConverter.ToInt32(infoData.GetRange(index, 4).ToArray(), 0);
+                        index += 4;
+                        output.AddRange(infoData.GetRange(index, numBytes));

```

Comment:
>output.AddRange(infoData.GetRange(index, numBytes)); [](http://example.com/codeflow?start=24&length=52)

fyi using the v5 test file included in the PR to run the test, have below error at this line for Block4.

-  $exception {"Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection."} System.ArgumentException

   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
   at Microsoft.CodeAnalysis.BinaryParsers.Dwarf.Read.AttributeValue(List`1 infoData, Int32& index, Attribute attribute, Int32 cuId) in C:\GH\binskim\src\BinaryParsers\ELFBinary\Dwarf\Read.cs:line 38

Reply:
all utility code are being replaced.